source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRView.mm

```diff
@@ -115,43 +116,124 @@ - (UIView *)render
-
+            [containerView addSubview:imgView];
+            [containerView sendSubviewToBack:imgView];
+            
+            [ACRView applyBackgroundImageConstraints:backgroundImageProperties imageView:imgView];
+            
             [imgView setContentHuggingPriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisHorizontal];
             [imgView setContentHuggingPriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisVertical];
             [imgView setContentCompressionResistancePriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisHorizontal];
             [imgView setContentCompressionResistancePriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisVertical];
         }
     }
-    [self callDidLoadElementsIfNeeded];
-    return newView;
+}
+
++ (void) applyBackgroundImageConstraints:(const AdaptiveCards::BackgroundImage *)backgroundImageProperties imageView:(UIImageView *)imageView
+{
+    UIView *parentView = [imageView superview];
+    UIImage *image = [imageView image];
+    switch (backgroundImageProperties->GetMode()) {
+        case AdaptiveCards::BackgroundImageMode::Repeat:
+        {
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeCenterX relatedBy:NSLayoutRelationEqual toItem:parentView attribute:NSLayoutAttributeCenterX multiplier:1.0 constant:0].active = YES;
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeCenterY relatedBy:NSLayoutRelationEqual toItem:parentView attribute:NSLayoutAttributeCenterY multiplier:1.0 constant:0].active = YES;
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:parentView attribute:NSLayoutAttributeTop multiplier:1.0 constant:0].active = YES;
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:parentView attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0].active = YES;
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeLeading relatedBy:NSLayoutRelationEqual toItem:parentView attribute:NSLayoutAttributeLeading multiplier:1.0 constant:0].active = YES;
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeTrailing relatedBy:NSLayoutRelationEqual toItem:parentView attribute:NSLayoutAttributeTrailing multiplier:1.0 constant:0].active = YES;
+            
+            imageView.contentMode = UIViewContentModeScaleAspectFill;
+            break;
+        }
+        case AdaptiveCards::BackgroundImageMode::RepeatHorizontally:
+        {
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:nil attribute:NSLayoutAttributeNotAnAttribute multiplier:1.0 constant:image.size.height].active = YES;
```

Comment:
this wouldn't work because height may not available when this function. This applies to all constraints that requires properties from UIImage which is loaded asynchronously.

as discussed previously, you will have to implement ACRIKVONotificationHandler protocol in the renderers that use background image, then apply this constraints in the implemented protocol.  You are very close.